File: How Many Numbers Are Smaller Than the Current Number.cpp
class Solution {
public:
    vector<int> smallerNumbersThanCurrent(vector<int>& nums) {
        int n=nums.size();
        vector <int> ans;
        int value=0;
      
        for(int i=0;i<n;i++){ 
             for(int j=0;j<n;j++)
             {   
               if(i!=j && nums.at(i)>nums.at(j))
                     value++;
             }
                    ans.push_back(value);
                    value=0;
        }
    return ans;
    }
};

File: progress sheet/how-many-numbers-are-smaller-than-the-current-number.py
class Solution:
    def smallerNumbersThanCurrent(self, nums: List[int]) -> List[int]:
        counts = Counter(nums)
        cumulative_sum = {num: sum(counts[key] for key in counts if key < num) for num in nums}
        ans = [cumulative_sum[num] for num in nums]
        return ans